plugins/modules/cloud/misc/terraform.py

```diff
@@ -449,12 +474,67 @@ def main():
     if state == 'present' and module.params.get('parallelism') is not None:
         command.append('-parallelism=%d' % module.params.get('parallelism'))
 
-    variables_args = []
-    for k, v in variables.items():
-        variables_args.extend([
-            '-var',
-            '{0}={1}'.format(k, v)
-        ])
+    def process_args(variables, top):
+        variables_args = []
+        lowlevel_out = []
+        for k, v in variables.items():
+            if top:
+                if (isinstance(v, dict)):
```

Comment:
Same below, this isn't C :-)
```suggestion
                if isinstance(v, dict):
```